### PR TITLE
[Resources.Container] Rename Container detector namespace

### LINKED ISSUE #1610: Reconsider `ResourceDetectors` namespaces and usages
# Issue that does not fit into other categories

**What are you trying to achieve?**

Each resource detector package today has a unique namespace. This results in substantial noise when including multiple detectors in a project, a situation which is fairly common. 

Here is an example of a recent change on one of our projects:
image

This also goes against the design of other components such as instrumentations and exporters, which follow the main namespaces of their respective metric type. Eg:

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/01c225798cefa2fe89637b78815c0a6a0c6b4c7b/src/OpenTelemetry.Instrumentation.AspNet/MeterProviderBuilderExtensions.cs#L8-L20

The main difference between these 2 is that detectors don't come with custom extension methods on `ResourceBuilder`, while other components are usually added via extensions on `TracerProviderBuilder` and `MeterProviderBuilder`.

What do you expect to see?

I'd like to see one of 3 approaches here:
1. Have all `OpenTelemetry.ResourceDetectors.*` packages have a root namespace of `OpenTelemetry.ResourceDetectors`
2. Have all `OpenTelemetry.ResourceDetectors.*` packages have a root namespace of `OpenTelemetry.Resources`
3. Have all `OpenTelemetry.ResourceDetectors.*` packages provide `.Add{detector_name}Detector` extension methods on top of `ResourceBuilder` and targeting namespace `OpenTelemetry.Resources`

From a consistency standpoint, I believe 3 would be the best as it stands today. After that, I like 2 as it doesn't introduce a new namespace, and then 3 as it at least unifies all of them in a single namespace.

The caveat with 2 and 1 is that they would potentially also apply to all other projects in a similar way, so it would make sense to apply them more broadly than just for Resource Detector packages.

Additionally, opting for 2 could prompt a different naming scheme altogether which I think would be more fitting, from "ResourceDetector" to just "Resources":
- `OpenTelemetry.ResourceDetectors.Host` -> `OpenTelemetry.Resources.Host`

**Additional context.**

I believe the current approach is extremely noisy and won't scale well as even more detectors are introduced. Additionally, even without regard to the noise issue, I dislike very granular namespaces like this and am fond of the pattern of "child" assemblies _NOT_ including the last segment of the assembly name in the final namespace. This is a common approach in many libraries, for example
`Microsoft.Extensions.DependencyInjection` vs `Microsoft.Extensions.DependencyInjection.Abstractions`, where the "Abstractions" in the assembly name does not participate in the final namespace of the classes inside:

image

### PR DESCRIPTION
**Pull request by joegoldman2**
Fixes #1610.

## Changes

- Renames OpenTelemetry.ResourceDetectors.Container to OpenTelemetry.Resources.Container
- Makes detector internal
- Adds `AddContainerDetector` extension method for ResourceBuilder

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by Kielek — APPROVED** [review]
LGTM

**Review by rajkumar-rangaraj — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -431,6 +420,17 @@ jobs:
...
+  build-test-resources-container:
+    needs: detect-changes
+    if: |
+      contains(needs.detect-changes.outputs.changes, 'resources-container')
+      || contains(needs.detect-changes.outputs.changes, 'build')
+      || contains(needs.detect-changes.outputs.changes, 'shared')
+    uses: ./.github/workflows/Component.BuildTest.yml
+    with:
+      project-name: Component[OpenTelemetry.Resources.Container]
+      code-cov-name: ResourceDetectors.Container
```
Comment (iskiselev) [inline .github/workflows/ci.yml:432]: Should it be renamed to Resources.Container?

**Review by iskiselev — CHANGES_REQUESTED** [review]
Looks good except minor missed rename. After we merge it, somebody would be required to mark as deprecated https://www.nuget.org/packages/OpenTelemetry.ResourceDetectors.Container pointing to a new name, and probably update recommended replacement in https://www.nuget.org/packages/OpenTelemetry.Extensions.Docker

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Kielek) [reply .github/workflows/ci.yml]: Good catch. Fixed in e87aa684ab11485c7ea9349b22e20a29537ba466 with couple other things

**Comment by Kielek** [comment]
> Looks good except minor missed rename. After we merge it, somebody would be required to mark as deprecated https://www.nuget.org/packages/OpenTelemetry.ResourceDetectors.Container pointing to a new name, and probably update recommended replacement in https://www.nuget.org/packages/OpenTelemetry.Extensions.Docker

I will handle nuget part, probably today, when all resource packages are released.

**Pull request merged**